packages/core/src/queries/log.ts

```diff
@@ -1,33 +1,54 @@
-import type { Log } from '@logto/schemas';
-import { token, Logs } from '@logto/schemas';
+import { type Log, LogResult, token, Logs } from '@logto/schemas';
 import { conditionalSql, convertToIdentifiers } from '@logto/shared';
 import type { CommonQueryMethods } from 'slonik';
 import { sql } from 'slonik';
+import { nativeEnum, object, string, type z } from 'zod';
 
 import { buildFindEntityByIdWithPool } from '#src/database/find-entity-by-id.js';
 import { buildInsertIntoWithPool } from '#src/database/insert-into.js';
 
 const { table, fields } = convertToIdentifiers(Logs);
 
-export type LogCondition = {
-  logKey?: string;
-  applicationId?: string;
-  userId?: string;
-};
+export const logConditionGuard = object({
```

Comment:
should we define guard exclusively based on different product usecases? e.g. user log won't have a hookId field. Should not put everything together?